public/app/plugins/panel/status-history/StatusHistoryTooltip2.tsx

```diff
@@ -0,0 +1,108 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import {
+  DataFrame,
+  Field,
+  formattedValueToString,
+  getDisplayProcessor,
+  getFieldDisplayName,
+  GrafanaTheme2,
+  TimeZone,
+  LinkModel,
+} from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
+import { VizTooltipContent } from '@grafana/ui/src/components/VizTooltip/VizTooltipContent';
+import { VizTooltipFooter } from '@grafana/ui/src/components/VizTooltip/VizTooltipFooter';
+import { VizTooltipHeader } from '@grafana/ui/src/components/VizTooltip/VizTooltipHeader';
+import { LabelValue } from '@grafana/ui/src/components/VizTooltip/types';
+
+interface StatusHistoryTooltipProps {
+  data: DataFrame[];
+  dataIdxs: Array<number | null>;
+  alignedData: DataFrame;
+  seriesIdx: number | null | undefined;
+  timeZone: TimeZone;
+  isPinned: boolean;
+}
+
+function fmt(field: Field, val: number): string {
+  if (field.display) {
+    return formattedValueToString(field.display(val));
+  }
+
+  return `${val}`;
+}
+
+export const StatusHistoryTooltip2 = ({
+  data,
+  dataIdxs,
+  alignedData,
+  seriesIdx,
+  timeZone,
+  isPinned,
+}: StatusHistoryTooltipProps) => {
+  const styles = useStyles2(getStyles);
+
+  // @todo: check other dataIdx, it can be undefined or null in array
+  const datapointIdx = dataIdxs.find((idx) => idx !== undefined);
+
+  if (!data || datapointIdx == null) {
+    return null;
+  }
+
+  const xField = alignedData.fields[0];
+  const field = alignedData.fields[seriesIdx!];
+
+  const links: Array<LinkModel<Field>> = [];
+  const linkLookup = new Set<string>();
+
+  if (field.getLinks) {
+    const v = field.values[datapointIdx];
+    const disp = field.display ? field.display(v) : { text: `${v}`, numeric: +v };
+    field.getLinks({ calculatedValue: disp, valueRowIndex: datapointIdx }).forEach((link) => {
+      const key = `${link.title}/${link.href}`;
+      if (!linkLookup.has(key)) {
+        links.push(link);
+        linkLookup.add(key);
+      }
+    });
+  }
+
+  const fieldFmt = field.display || getDisplayProcessor();
+  const value = field.values[datapointIdx!];
+  const display = fieldFmt(value);
+
+  const getHeaderLabel = (): LabelValue => {
+    return {
+      label: '',
+      value: fmt(xField, xField.values[datapointIdx]),
+    };
+  };
+
+  const getContentLabelValue = () => {
+    return [
+      {
+        label: getFieldDisplayName(field),
+        value: fmt(field, field.values[datapointIdx]),
+        color: display.color,
+      },
+    ];
+  };
+
+  return (
+    <div className={styles.wrapper}>
+      <VizTooltipHeader headerLabel={getHeaderLabel()} />
+      <VizTooltipContent contentLabelValue={getContentLabelValue()} />
+      {isPinned && <VizTooltipFooter dataLinks={links} canAnnotate={false} />}
+    </div>
+  );
+};
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  wrapper: css({
+    display: 'flex',
+    flexDirection: 'column',
+    width: '280px',
```

Comment:
What is driving this number?

Reply:
This works as expected locally and is a great improvement, good job! LGTM other than the one question I have regarding width. If this is a standard number we intend to use elsewhere, I suggest pulling it out as a shared const somewhere. Otherwise, is there a more responsive way to do this?

Reply:
It is the fixed width of the tooltip. It's used to define every tooltip width, so probably we can put it as a constant somewhere.
I'm also totally fine to leave it just like that 😁
@adela-almasan what do you think?